components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,215 @@
+      return rows.map((row) => row.column_name);
+    },
+    /**
+     * Gets a table's primary key field
+     * @param {string} table - Name of the table to get the primary key for
+     * @returns Name of the primary key column
+     */
+    async getPrimaryKey(table) {
+      const rows = await this.executeQuery(`SELECT a.attname FROM pg_index i JOIN pg_attribute a ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) WHERE i.indrelid = '${table}'::regclass AND i.indisprimary`);
+      return rows[0].attname;
+    },
+    /**
+     * Gets rows in a table with values greater than lastResult
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Column to filter by lastResult
+     * @param {string} [lastResult] - A column value. Used to retrieve new rows since the
+     *  last time the table was queried.
+     * @returns Array of rows returned from the query
+     */
+    async getRows(table, column, lastResult = null) {
+      const query = `SELECT * FROM ${table} ${lastResult
+        ? "WHERE " + column + ">" + lastResult
+        : ""} ORDER BY ${column} DESC`;
+      return this.executeQuery(query);
+    },
+    /**
+     * Gets a single row from a table based on a lookup column & value
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Column to filter by value
+     * @param {string} value - A column value to search for
+     * @returns A single database row
+     */
+    async findRowByValue(table, column, value) {
+      const rows = await this.executeQuery(`SELECT * FROM ${table} WHERE ${column} = '${value}'`);
+      return rows[0];
+    },
+    /**
+     * Gets the number of rows in a table
+     * @param {string} table - Name of database table to query
+     * @returns Count of how many rows exist in the table
+     */
+    async getRowCount(table) {
+      const response = await this.executeQuery(`SELECT COUNT(*) FROM ${table}`);
+      return response[0].count;
+    },
+    /**
+     * Deletes a row or rows in a table based on a lookup column & value
+     * @param {string} table - Name of database table to delete from
+     * @param {string} column - Column used to find the row(s) to delete
+     * @param {string} value - A column value. Used to find the row(s) to delete
+     */
+    async deleteRows(table, column, value) {
+      await this.executeQuery(`DELETE FROM ${table} WHERE ${column} = '${value}'`);
+    },
+    /**
+     * Inserts a row in a table
+     * @param {string} table - Name of database table to insert row into
+     * @param {array} columns - Array of column names
+     * @param {array} values - Array of values corresponding to the column names provided
+     * @returns The newly created row
+     */
+    async insertRow(table, columns, values) {
+      return this.executeQuery(`INSERT INTO ${table} (${columns}) VALUES (${values.map((v) => "'" + v + "'")}) RETURNING *`);
+    },
```

Comment:
If you change `executeQuery` the way I suggested previously I would refactor this one to
```suggestion
    async insertRow(table, columns, values) {
      const placeholders = this.getPlaceholders({
        values,
      });
      return this.executeQuery({
        text: format(`
          INSERT INTO %I (${columns}) 
            VALUES (${placeholders})
            RETURNING *
        `, table),
        values,
      });
    },
```

with this additional function which will give you the necessary placeholder values for the query
```javascript
    getPlaceholders({
      values = [], fromIndex = 1,
    }) {
      return values.map((_, index) => `$${index + fromIndex}`);
    },
```